### ISSUE TITLE
[ENH] Draft classification eval suite and lightweight classifier exploration

### ISSUE THREAD

**Issue by ksharma6** [ksharma6/inbox_zero#41]
**Is your feature request related to a problem? Please describe.**
The current `_process_emails_for_drafts` node uses the same LLM call to both classify whether an email warrants a draft response and to perform the draft analysis. This has two problems
1) classification is a much simpler task than draft generation but pays the same latency and token cost
2) there is no eval harness to measure how accurately the system determines which emails warrant a response making it hard to know if a lighter approach would be safe to adopt.

**Describe the solution you'd like**
Three related pieces of work:

1. **Dedicated email dataset for classification eval**. Build a ground-truth dataset of synthetic `EmailMessage` objects specifically for evaluating draft-worthiness classification. Each entry should be wrapped in a typed `ClassificationCase` dataclass (fixture-layer only, not in the production model) that pairs the email with its expected label:

```python
@dataclass
class ClassificationCase:
    email: EmailMessage
    draft_warranted: bool
    label: str  # action_item | question | ooo | payment_confirmation | fyi
```

The dataset should cover at minimum:
- `action_item` — explicit request requiring a reply (draft warranted)
- `question` — direct question requiring an answer (draft warranted)
- `ooo` — automated out-of-office reply (no draft warranted)
- `payment_confirmation` — transactional notification (no draft warranted)
- `fyi` — informational, no response expected (no draft warranted)

The dataset lives in `tests/fixtures/email_datasets.py` and is versioned alongside the code so classification accuracy can be tracked over time as the implementation changes.
2. **Eval suite for draft classification.** Create a `tests/evals/test_draft_classification.py` module with a typed `ClassificationCase` wrapper (an `email: EmailMessage` paired with a `draft_warranted: bool` ground-truth label) and a dataset covering the five core cases: `action_item`, `question`, `ooo`, `payment_confirmation`, and `fyi`. The eval should assert a minimum accuracy threshold rather than exact match, and be gated behind a `@pytest.mark.eval` marker so it is excluded from the default CI run. 

3. **ADR: lightweight classifier.** Explore replacing the LLM classification gate with a cheaper mechanism — a smaller/faster model, a zero-shot classifier, or a heuristic using the cascade/routing pattern. The eval suite in (1) provides the measurement surface to determine whether accuracy holds. The ADR should document the cost, latency, and accuracy tradeoffs of each option against the current single-LLM approach, consistent with the format of ADR 001 and ADR 002.

**Describe alternatives you've considered**
- Keep classification inside the existing `_process_emails_for_drafts` LLM call. Simplest, but pays full model cost even for emails that are trivially non-actionable.
- Use prompt caching on the classification prompt. Reduces cost on repeated identical prompts but does not reduce latency on first call and does not simplify the prompt structure.
- Fine-tune a small model on inbox data. Highest accuracy ceiling but requires labelled training data and MLOps infrastructure not present in the current stack. Deferred.

**Additional context**
Related ADRs: [001 — Email Caching Strategy](docs/decisions/001-email-caching-strategy.md), [002 — Parallelisation Strategy](docs/decisions/002-parallelization-strategy.md). The parallelisation ADR already identifies draft generation as the dominant latency contributor; a classification gate that skips the draft step entirely for non-actionable emails compounds that win.
